azext_edge/edge/providers/check/base.py

```diff
@@ -712,6 +719,81 @@ def filter_by_namespace(resources: List[dict], namespace: str) -> List[dict]:
     return [resource for resource in resources if get_resource_namespace(resource) == namespace]
 
 
+def process_dict_resource(
+    check_manager: CheckManager,
+    target_name: str,
+    resource: dict,
+    namespace: str,
+    padding: int,
+    prop_name: Optional[str] = None
+) -> None:
+    if prop_name:
+        check_manager.add_display(
+            target_name=target_name,
+            namespace=namespace,
+            display=Padding(f"{prop_name}:", (0, 0, 0, padding))
+        )
+        padding += 4
+    for key, value in resource.items():
+        if isinstance(value, dict):
+            check_manager.add_display(
+                target_name=target_name,
+                namespace=namespace,
+                display=Padding(f"{key}:", (0, 0, 0, padding))
+            )
+            process_dict_resource(
+                check_manager=check_manager,
+                target_name=target_name,
+                resource=value,
+                namespace=namespace,
+                padding=padding + 4
+            )
+        elif isinstance(value, list):
+            if len(value) == 0:
+                continue
+
+            display_text = f"{key}:"
+            check_manager.add_display(
+                target_name=target_name,
+                namespace=namespace,
+                display=Padding(display_text, (0, 0, 0, padding))
+            )
+
+            for item in value:
+                process_dict_resource(
+                    check_manager=check_manager,
+                    target_name=target_name,
+                    resource=item,
+                    namespace=namespace,
+                    padding=padding + 4,
+                    prop_name=f"{key} {value.index(item) + 1}"
```

Comment:
Hi @c-ryan-k, you are right about the list format, here is an example:
<img width="157" alt="image" src="https://github.com/Azure/azure-iot-ops-cli-extension/assets/22055990/61d95863-2f69-4de5-bfe3-3a61ed17f63a">

Originally I decided to use typename + index instead of name because I was concerned about if there is certain data type without a name property. I do think using name as header is the best option, but if the name property does not exist, should we fall back to index? thanks!

Also seperate out a process_list_resource sounds good to me! thanks for the suggestion!

Reply:
updated according to suggestion, if there is no name found, fallback to index method